### PR TITLE
Check if is a resource before try set log rotate

### LINKED ISSUE #14544: fstat() expects parameter 1 to be resource, boolean given
FYI: This is seen from the PayPal settings page when first enabling the payment gateway and saving setting for the first time, the enable logging was checked. 

`fstat() expects parameter 1 to be resource, boolean given in /var/www/wc30/wp-content/plugins/woocommerce/includes/log-handlers/class-wc-log-handler-file.php on line 270`

### PR DESCRIPTION
**Pull request by claudiosanches**
Fixes #14544

### CODE REVIEW

File: includes/log-handlers/class-wc-log-handler-file.php [hunk 1]
```diff
@@ -266,7 +266,7 @@ public function remove( $handle ) {
 	protected function should_rotate( $handle ) {
 		$file = self::get_log_file_path( $handle );
 		if ( $file ) {
-			if ( $this->is_open( $handle ) ) {
+			if ( $this->is_open( $handle ) && is_resource( $this->handles[ $handle ] ) ) {
```
Comment (sirreal) [inline includes/log-handlers/class-wc-log-handler-file.php]: Do you think `is_resource` should be added to the `is_open` method? `is_open` could prune handles that aren't resources, it looks like most calls are `is_open && is_resource`.

Comment (claudiosanches) [inline includes/log-handlers/class-wc-log-handler-file.php]: @sirreal I just copied your own code. There is other example of `$this->is_open( $handle ) && is_resource( $this->handles[ $handle ] )` in the close method.
And if `is_open` is able to prune such a thing he should not seen any error already: https://github.com/woocommerce/woocommerce/issues/14544

So should I change the `is_open` method?

File: includes/log-handlers/class-wc-log-handler-file.php [hunk 2]
```diff
@@ -163,7 +163,7 @@ protected function open( $handle, $mode = 'a' ) {
 	 * @return bool True if $handle is open.
 	 */
 	protected function is_open( $handle ) {
-		return array_key_exists( $handle, $this->handles );
+		return array_key_exists( $handle, $this->handles ) && is_resource( $this->handles[ $handle ] );
```
Comment (sirreal) [inline includes/log-handlers/class-wc-log-handler-file.php]: This seems a sensible thing to check here.

**Comment by sirreal** [comment]
It looks like there's actually a bug here:

https://github.com/woocommerce/woocommerce/blob/dfcbeaf044e6cc8b10c849c75ecdabaa19a8768a/includes/log-handlers/class-wc-log-handler-file.php#L151-L153

```php
if ( $this->handles[ $handle ] = @fopen( $file, $mode ) ) {
	return true;
}
```

Rather than sticking the result directly into `$this->handles`, we need to do the false test first, which is how a boolean is sneaking into `$this->handles`. There should be an intermediate check something like the following:

```php
if ( $resource = @fopen( $file, $mode ) ) {
	$this->handles[ $handle ] = $resource;
	return true;
}
```

**Comment by claudiosanches** [comment]
@sirreal looks good now? Or I should remove the check in `is_open()`?

**Review by sirreal — APPROVED** [review]
LGTM. Thanks for fixing this 👍

**Pull request merged**